src/scheduler/PCB.java

```diff
@@ -97,13 +97,30 @@ public void setProcessQuantumTo1() {
 		this.processQuantum = 1;
 	}
 
-	@Override
-	public int compareTo(PCB pcb) {
+	public int compareToForInitialization(PCB pcb) {
+		int condition = commonConditions(pcb);
+		if (condition != 0) return condition;
+		else {
+			if (this.process.getNumber() > pcb.process.getNumber()) return 1;
```

Comment:
Here we should compare `process.getName`, not number, as our professor told us to do.